core/domain/html_validation_service.py

```diff
@@ -705,3 +705,156 @@ def convert_svg_diagram_tags_to_image_tags(html_string):
         tabs['tab_contents-with-value'] = escape_html(
             json.dumps(tab_content_list))
     return python_utils.UNICODE(soup)
+
+def fix_incorrectly_encoded_chars(html_string):
+    """Replaces incorrectly encoded character with the correct one in a given
+    HTML string.
+
+    Args:
+        html_string: str. The HTML string to modify.
+
+    Returns:
+        str. The updated html string.
+    """
+    soup = bs4.BeautifulSoup(
```

Comment:
I think this function does not handle html in tabs/collapsibles correctly.

Reply:
You're right. Fixed it now!